### DISCUSSION TITLE
OpenDeck X(X)L Board Question

### DISCUSSION THREAD

**Discussion by OhNoItSJoe** [shanteacontrols/OpenDeck#323] [General]
Hello Shantea!

Upon searching for a MIDI-Controller to MIDIfy an old Organ to use it with Haupwerk and GrandOrgue I stumbled upon your project.

The organ I want to bring into the digital age has a "few" buttons tho.

![orgel](https://github.com/user-attachments/assets/9b871d2c-87ef-460d-9fac-c23add528ef6)

183 keys, 25 Pedals, 60 Register buttons, 48 at the bottom and like 40 more, don't have it with me, yet.

Since I need more inputs, I'd like if its possible to simply increase the size of the shift registers of the L board to get more inputs. My estimations are 5-6 registers in total instead of the existing 2.

If I understood your circuit right, it would be as simple as adding more 74HC165 and daisy-chain them.

Since I also need more LEDs (I want to add a light to each key), I'd also have to quadruple the MAX2719, as far as I understood it's quite simple here, too since it also supports daisy-chaining.

Is this possible or am I forgetting something, maybe there are limitations of the Pico?

In theory I could also use 4 L boards for my project but I'd rather have one board.

**Comment by OhNoItSJoe** [comment]
I've updated the schematics to represent the changes I'd like to do for my purposes.
Would this work?

Since I'm new to KiCAD, if someone wants to help me with setting up the PCB, help is appreciated.
Of course you'd get a sample of the board once I got it :)
[opendeck_xxl.zip](https://github.com/user-attachments/files/18267484/opendeck_xxl.zip)

**Comment by laenzlinger** [comment]
what kind of help do you need?

**Reply by gedicke-dev** [reply]
I'm not really sure if this what I have in mind would work.
But I also found a chip that could work for my idea that provides 64 Inputs via I2C.

https://www.lumissil.com/applications/industrial/appliance/major-appliances/range-hood/is31io7326

I think this would be a better way to accomplish, I'm in touch with the company and will check if it can register enough keystrokes at the same time to function for an organ.
But 4 of those chips seem a bit easier to implement.

**Comment by laenzlinger** [comment]
Ok contact me if you need help.

**Reply by OhNoItSJoe** [reply]
@laenzlinger would appreciate your help! Should we hop into a call and discuss what I've planned and then share the results here?
I am getting the organ console soon.
I have some experience with electrical circuits, but never manufactured my own PCB, so would be more than glad for your help!
